kitsune/sumo/utils.py

```diff
@@ -315,6 +319,42 @@ def get_browser(user_agent):
     return browser
 
 
+def has_blocked_link(data):
+    # TODO: deal with punycode when presented in non-ascii form
+    for match in POTENTIAL_LINK_REGEX.finditer(data):
+        tld = match.group(1).upper()
+        if tld in VALID_TLDS:
+            full_domain = match.group(0).lower()
+            in_allowlist = False
+            for allowed_domain in settings.ALLOW_LINKS_FROM:
+                split = full_domain.rsplit(allowed_domain, 1)
+                if len(split) != 2:
+                    # allowed_domain isn't in full_domain, or something went wrong
+                    continue
+                if split[-1] != '':
+                    # allowed_domain isn't at the end of full_domain
+                    continue
+                if split[0] == '':
+                    # allowed_domain equals full_domain
+                    in_allowlist = True
+                    break
```

Comment:
If you break then you will allow a post that maybe has more links in it that are not whitelisted.

Reply:
This breaks the `for allowed_domain in settings.ALLOW_LINKS_FROM` loop, not the `for match in POTENTIAL_LINK_REGEX.finditer(data)` one

This testcase covers the situation you're describing:
https://github.com/mozilla/kitsune/blob/2e66162243caeb1594db463c215f9c7c27f4351c/kitsune/sumo/tests/test_utils.py#L230